numpy/lib/arraypad.py

```diff
@@ -779,11 +779,11 @@ def pad(array, pad_width, mode='constant', **kwargs):
     }
     try:
         unsupported_kwargs = set(kwargs) - set(allowed_kwargs[mode])
-    except KeyError:
-        raise ValueError("mode '{}' is not supported".format(mode))
+    except KeyError as e:
+        raise ValueError("mode '{}' is not supported".format(mode)) from e
```

Comment:
This should be `from None`

Reply:
Just to provide stack trace I thought from KeyError would be a better approach. Anyways updated pull request you can review it now. Thanks for your comments.

Reply:
The stack trace isn't interesting though, because all the user needs to know is they passed an illegal `mode`.